### PR TITLE
Add PublicKeyProvider interface

### PR DESCRIPTION
**Pull request by bobcallaway**
Given the requirements of cosign, I split out the PublicKey() method into its own interface such that it can be implemented by both signers and verifiers (since both have a public key).

Also small fixes including:
 - ed25519 passes by value rather than pointer for public/private keys
 - adds SupportedAlgorithms() and DefaultAlgorithm() method for KMS SignerVerifiers (ideal for use in CreateKey() method)

Signed-off-by: Bob Callaway <bob.callaway@gmail.com>

### CODE REVIEW

**Comment by bobcallaway** [comment]
@dekkagaijin could you please take a look? thanks!

File: pkg/signature/verifier.go [hunk 1]
```diff
@@ -43,8 +44,8 @@ func LoadVerifier(publicKey crypto.PublicKey, hashFunc crypto.Hash) (Verifier, e
 		return LoadRSAPKCS1v15Verifier(pk, hashFunc)
 	case *ecdsa.PublicKey:
 		return LoadECDSAVerifier(pk, hashFunc)
-	case *ed25519.PublicKey:
-		return LoadED25519Verifier(*pk)
+	case ed25519.PublicKey:
```
Comment (lukehinds) [inline pkg/signature/verifier.go:47]: I guess you meant to remove the pointer here (line 47)?

**Review by lukehinds — APPROVED** [review]
LGTM, just a small note.

File: pkg/signature/verifier.go (hunk 1 above)
Comment (bobcallaway) [reply pkg/signature/verifier.go:47]: yes, this was intentional. `rsa` and `ecdsa` represent their public key in structs, but `ed25519` just has a `[]byte` representation so they're handled differently.

**Pull request merged**